DigitalLearningSolutions.Web/Controllers/Register/RegisterDelegateByCentreController.cs

```diff
@@ -0,0 +1,195 @@
+
+            ValidateEmailAddress(model);
+
+            return View(model);
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationByCentreData>))]
+        [HttpPost]
+        [Route("PersonalInformation")]
+        public IActionResult PersonalInformation(PersonalInformationViewModel model)
+        {
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            ValidateEmailAddress(model);
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            data.SetPersonalInformation(model);
+            TempData.Set(data);
+
+            return RedirectToAction("LearnerInformation");
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationByCentreData>))]
+        [HttpGet]
+        [Route("LearnerInformation")]
+        public IActionResult LearnerInformation()
+        {
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            var model = new LearnerInformationViewModel(data);
+
+            PopulateLearnerInformationExtraFields(model, data);
+
+            return View(model);
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationByCentreData>))]
+        [HttpPost]
+        [Route("LearnerInformation")]
+        public IActionResult LearnerInformation(LearnerInformationViewModel model)
+        {
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            var centreId = data.Centre!.Value;
+
+            customPromptHelper.ValidateCustomPrompts(
+                centreId,
+                model.Answer1,
+                model.Answer2,
+                model.Answer3,
+                model.Answer4,
+                model.Answer5,
+                model.Answer6,
+                ModelState
+            );
+
+            if (!ModelState.IsValid)
+            {
+                PopulateLearnerInformationExtraFields(model, data);
+                return View(model);
+            }
+
+            data.SetLearnerInformation(model);
+            TempData.Set(data);
+
+            return new OkResult();
+        }
+
+        private void SetCentreDelegateRegistrationData(int centreId)
+        {
+            var centreDelegateRegistrationData = new DelegateRegistrationByCentreData
+            {
+                IsCentreSpecificRegistration = true,
+                Centre = centreId
+            };
+            var id = centreDelegateRegistrationData.Id;
+
+            Response.Cookies.Append(
+                CookieName,
+                id.ToString(),
+                new CookieOptions
+                {
+                    Expires = DateTimeOffset.UtcNow.AddDays(30)
+                }
+            );
+
+            TempData.Set(centreDelegateRegistrationData);
+        }
+
+        private void ValidateEmailAddress(PersonalInformationViewModel model)

```

Comment:
This method seems to be identical to the one in the RegisterController. Might be worth refactoring it into a helper method that returns true/false, and then the methods that call it can add the model error.